### ISSUE TITLE
Use same metric for early stopping and saving best model

### ISSUE THREAD

**Issue by johann-petrak** [GateNLP/gate-lf-pytorch-json#42]
The parameter --es_metric allows to choose accuracy or loss for the early stopping decision, but the best model is always saved based on the best validation accuracy. We should use the same metric there as for early stopping.
